### DISCUSSION TITLE
my workflow is not triggered while creating the rule also

### DISCUSSION THREAD

**Discussion by eshwar93** [StackStorm/st2#5966] [Workflows]
I have created the rule to trigger my workflow when ever the message send the kafka-topic
{
  "tags": [],
  "uid": "rule:st2test:AU_test",
  "metadata_file": "",
  "name": "AU_test",
  "ref": "st2test.AU_test",
  "description": "test the data",
  "pack": "st2test",
  "type": {
    "ref": "standard",
    "parameters": {}
  },
  "trigger": {
    "type": "kafka.new_message",
    "parameters": {},
    "ref": "kafka.new_message",
    "description": "Trigger which indicates that a new message has arrived to Apache Kafka"
  },
  "criteria": {
    "trigger.topic": {
      "type": "equals",
      "pattern": "sample-test"
    }
  },
  "action": {
    "ref": "st2test.test_wf",
    "parameters": {
      "run": "test",
      "notify": [
        6
      ]
    },
    "description": "execute the flow"
  },
  "context": {
    "user": "st2admin"
  },
  "enabled": true,
  "id": "6445443e757510e5c70f6923"
}

**Comment by nzlosh** [comment]
Try working through the troubleshooting section for rules not matching https://docs.stackstorm.com/troubleshooting/rules.html
